log/field.go

```diff
@@ -198,3 +201,41 @@ func (lf Field) Marshal(visitor Encoder) {
 		visitor.EmitLazyLogger(lf.interfaceVal.(LazyLogger))
 	}
 }
+
+// Key returns the field's key.
+func (lf Field) Key() string {
+	return lf.key
+}
+
+// Value returns the field's value as interface{}.
+func (lf Field) Value() interface{} {
+	switch lf.fieldType {
+	case stringType:
+		return lf.stringVal
+	case boolType:
+		return lf.numericVal != 0
+	case intType:
+		return int(lf.numericVal)
+	case int32Type:
+		return int32(lf.numericVal)
+	case int64Type:
+		return int64(lf.numericVal)
+	case uint32Type:
+		return uint32(lf.numericVal)
+	case uint64Type:
+		return uint64(lf.numericVal)
+	case float32Type:
+		return math.Float32frombits(uint32(lf.numericVal))
+	case float64Type:
+		return math.Float64frombits(uint64(lf.numericVal))
+	case errorType, objectType, lazyLoggerType:
```

Comment:
Ah.. Yeah I guess we would need to define an Encoder that captures the emitted key-value, and run the lazy logger. I'll file an issue.

BTW this makes me curious - why does the `Encoder` interface have an `EmitLazyLogger` method? Wouldn't any implementation of that (that follows the intended purpose) always just call the lazyLogger? Seems that instead of calling `visitor.EmitLazyLogger(lf.interfaceVal.(LazyLogger))` we could call `lf.interfaceVal.(LazyLogger)(visitor)` no?